blueprints/multi-interface-servers.md

```diff
@@ -0,0 +1,666 @@
+Indexes:
+    "interface_pkey" PRIMARY KEY, btree (name, server)
+Foreign-key constraints:
+    "interface_server_fkey" FOREIGN KEY (server) REFERENCES server(id)
+```
+Another new table will need to be created to contain address information, which
+is described below.
+```text
+                 Table "public.ip_address"
+     Column     |  Type   | Collation | Nullable | Default
+----------------+---------+-----------+----------+---------
+ address        | inet    |           | not null |
+ gateway        | inet    |           | not null |
+ serviceaddress | boolean |           | not null |
+ server         | bigint  |           | not null |
+ interface      | text    |           | not null |
+Indexes:
+    "ip_address_pkey" PRIMARY KEY, btree (address, interface, server)
+Check constraints:
+    "ip_address_gateway_check" CHECK (masklen(gateway) = 0)
+Foreign-key constraints:
+    "ip_address_interface_fkey" FOREIGN KEY (interface, server) REFERENCES interface(name, server)
+    "ip_address_server_fkey" FOREIGN KEY (server) REFERENCES server(id)
+```
+
+This should be sufficient to capture the model changes. In addition to creating
+these tables, a migration will need to be written to convert currently stored
+server information to utilize these new tables.
+
+#### API
+<a name="sec:api"></a>
+The affected endpoints will be:
+
+##### `/cdns/{{name}}/configs/monitoring`
+The current payload includes a `trafficServers` property that contains an array
+of cache server representations to be monitored by a Traffic Monitor within the
+path-specified CDN. An example of one such array member is given below.
+```json
+{
+	"cacheGroup": "Ciab_Edge",
+	"fqdn": "edge.infra.ciab.test",
+	"hashId": "edge",
+	"hostName": "edge",
+	"interfacename": "eth0",
+	"ip6": "dead::babe",
+	"ip": "192.168.0.0",
+	"port": 80,
+	"profile": "ATS_EDGE",
+	"status": "REPORTED",
+	"type": "EDGE"
+}
+```
+This will need to change to reflect the new set of interfaces, which for the
+above example with no additional interfaces would look like:
+```json
+{
+	"cacheGroup": "Ciab_Edge",
+	"fqdn": "edge.infra.ciab.test",
+	"hashId": "edge",
+	"hostName": "edge",
+	"interfaces": [
+		{
+			"name": "eth0",
+			"ipAddresses": [
+				{
+					"address": "192.168.0.0/32",
+					"serviceAddress": true
+				},
+				{
+					"address": "dead::babe/128",
+					"serviceAddress": true
+				}
+			]
+		}
+	],
+	"port": 80,
+	"profile": "ATS_EDGE",
+	"status": "REPORTED",
+	"type": "EDGE"
+}
+```
+Note that only interfaces that have "monitor" set to `true` should appear at all
+in the payloads of this endpoint. Further note that neither gateways nor
+<abbr title="Maximum Transmission Unit">MTU</abbr> are reported.
+
+##### `/cdns/{{name}}/snapshot`
+This is a semantic change only - payloads remain structurally unchanged. The
+reported `ip`, `ip6` and `interfaceName` fields for elements of the
+`contentServers` array property should be an arbitrarily chosen IPv4 service
```

Comment:
arbitrarily?

Reply:
It's not. The intention was to convey that a service-address-containing interface may or may not have an ipv4 or ipv6 address, and the mechanism for choosing one isn't specified in that section. But it needs to be worded better.